sdk/core/core/inc/az_json_data.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_JSON_DATA_H
+#define _az_JSON_DATA_H
+
+#include <az_json_token.h>
+#include <az_mut_span.h>
+#include <az_span.h>
+
+#include <stdbool.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_json_data az_json_data;
```

Comment:
Would it be possible to move az_json_array's declaration further down, so that we don't need to typedef it beforehand?

Reply:
circular dependencies